packages/engine/src/proto.rs

```diff
@@ -226,9 +227,27 @@ pub struct ExperimentRunBase {
 }
 
 impl ExperimentRunBase {
-    /// Returns `true` if the experiment uses Python init or has any Python behavior.
-    pub fn requires_python_runner(&self) -> bool {
-        self.project_base.package_init.initial_state.name == InitialStateName::InitPy
+    /// Returns a `RunnerSpawnConfig` matching the config required by the files present in the
+    /// experiment.
+    pub fn runner_spawn_config(&self) -> RunnerSpawnConfig {
+        RunnerSpawnConfig {
+            python: self.requires_runner(Language::Python),
+            rust: self.requires_runner(Language::Rust),
+            javascript: self.requires_runner(Language::JavaScript),
+        }
+    }
+
+    /// Returns `true` if the experiment uses the language's init or has any behavior of the
+    /// language.
+    fn requires_runner(&self, language: Language) -> bool {
+        let requires_init = match (language, &self.project_base.package_init.initial_state.name) {
```

Comment:
`match` is far more readable IMO, let's ignore clippy here.

Reply:
:smile: I was checking how it looked when I saw your comment.
Ingored clippy https://github.com/hashintel/hash/pull/552/commits/531df30ad1795a5a79ec23c75cca3a2b0fcba25e